app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/LearningTest.kt

```diff
@@ -1,49 +1,98 @@
 package com.github.HumanLearning2021.HumanLearningApp
 
-import android.view.View
-import androidx.test.espresso.Espresso.onView
-import androidx.test.espresso.UiController
-import androidx.test.espresso.ViewAction
-import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiObject
+import androidx.test.uiautomator.UiSelector
 import com.github.HumanLearning2021.HumanLearningApp.view.LearningActivity
-import org.hamcrest.Matcher
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.Assert.assertThat
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class LearningTest {
 
-    @Rule @JvmField // see https://stackoverflow.com/a/34558666/7158887 for the @JvmField
+    @get:Rule
     val testRule = ActivityScenarioRule(LearningActivity::class.java)
 
-    @Test
-    fun dragImageOnCorrectCategory(){
+    val NUMBER_OF_CATEGORIES = 3
+    val CATEGORY_VIEW_CLASS_NAME = "android.widget.ImageView"
+    val NUMBER_OF_DRAG_STEPS = 10
 
-        val from: View // TODO
-        val to : View // TODO
-        onView(withId(R.id.learning_im_to_sort))
-                .perform(DragAndDropAction(to))
+    lateinit var mDevice: UiDevice
+    @Before fun init(){
+        mDevice = UiDevice.getInstance(getInstrumentation())
     }
 
-    class DragAndDropAction (val to: View) : ViewAction {
-        override fun getConstraints(): Matcher<View> {
-            TODO("Not yet implemented")
-        }
-
-        override fun getDescription(): String {
-            TODO("Not yet implemented")
+    @Test fun dragImageOnCorrectCategory(){
+        val imToSort = getImageToSort()
+        val startDescr = imToSort.contentDescription
+        val NUMBER_OF_TENTATIVES = 100
```

Comment:
```suggestion
        val NUMBER_OF_ATTEMPTS = 100
```
"tentative" is an improper gallicism :fr: in English, _tentative_  is an adjective. _attempts_ is a better alternative

Reply:
indeed, I didn't remember the proper english translation. Thanks

Reply:
(also the comment)

You're welcome